### PR TITLE
WIP: Try updating to actions/cache@v2 and cargo-culting rustup action

### PR DESCRIPTION
**Pull request by ispivey**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -42,35 +42,34 @@ jobs:
...
 
       - name: Cache Cargo registry
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
           path: ~/.cargo/registry
-          key: ${{ matrix.build }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            ${{ matrix.build }}-cargo-registry-
+          key: ${{ matrix.build }}-cargo-registry-beep-${{ hashFiles('**/Cargo.lock') }}
```
Comment (EverlastingBugstopper) [inline .github/workflows/test.yml:48]: beep?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -42,35 +42,34 @@ jobs:
...
-            ${{ matrix.build }}-target-
 
       - name: Install Rust
         run: |
           rustup update ${{ matrix.rust }} --no-self-update
           rustup default ${{ matrix.rust }}
 
       - name: Run Tests
-        run: cargo test --features "strict"
+        run: cargo test -vv --features "strict"
```
Comment (EverlastingBugstopper) [inline .github/workflows/test.yml:68]: i... dont think we want `-vv` (very verbose) here. it prints out a bunch of junk from dependencies at the start of the command.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -42,35 +42,34 @@ jobs:
...
         env:
           RUST_BACKTRACE: 1
+
+      - name: Clear the cargo caches
+        run: |
+          cargo install cargo-cache --no-default-features --features ci-autoclean
+          cargo-cache
```
Comment (EverlastingBugstopper) [inline .github/workflows/test.yml:75]:
```suggestion
          cargo cache
```

**Comment by ispivey** [comment]
@EverlastingBugstopper sorry, this is very very draft – just had to make a draft PR in order to get CI to run so I could play with it. Not really ready for meaningful review at this point (tho I actually learned things, so I appreciate it 😁 )

File: .github/workflows/test.yml (hunk 1 above)
Comment (ispivey) [reply .github/workflows/test.yml:48]: just my low-budget cache-busting while I was playing around with caching

**Comment by EverlastingBugstopper** [comment]
gonna close this - feel free to reopen if it's something you wanna revisit

**Pull request closed without merging**